prow/cmd/checkconfig/main.go

```diff
@@ -40,7 +41,46 @@ type options struct {
 	jobConfigPath string
 	pluginConfig  string
 
-	strict bool
+	warnings    warningFlags
+	allWarnings bool
+	strict      bool
+}
+
+func (o *options) reportWarning(err error, msg string) {
+	if o.strict {
+		logrus.WithError(err).Fatal(msg)
+	}
+	logrus.WithError(err).Warn(msg)
+}
+
+type warningFlags []string
```

Comment:
This is pretty much the same as what is implemented in the `flagutil` package: https://github.com/kubernetes/test-infra/blob/master/prow/flagutil/flagutil.go

Reply:
Nifty, will refactor to use that